tasks/need_fixing_linters/golangci_lint_parser.py

```diff
@@ -0,0 +1,125 @@
+
+CODEOWNERS_FILE_PATH = ".github/CODEOWNERS"
+
+def read_owners(owners_file: str):
+    """
+    Parses the CODEOWNERS file.
+    """
+    with open(owners_file, 'r', encoding="utf-8") as f:
+        codeowners = CodeOwners(f.read())
+    return codeowners
+
+
+def parse_file(golangci_lint_output: str):
+    """
+    Parses the output of the golangci-lint run.
+    Returns a Dict(linter: List(base_path, row, col, lint, descr)).
+    """
+    lints = []
+    current_module = None
+    for line in golangci_lint_output.split("\n"):
+        line = line.strip()
+
+        match = re.match(MODULE_PATTERN, line)
+        if match:
+            module_path, flavor = match.groups()
+            current_module = module_path
+            continue
+
+        if current_module is None:
+            continue
+
+        match = re.match(LINT_PATTERN, line)
+        if match is None:
+            continue
+
+        file_path, row, col, lint, _, descr, linter = match.groups()
+        full_path = os.path.normpath(os.path.join(current_module, file_path))
+        base_path = full_path.removeprefix(os.getcwd()).removeprefix('/')
+        lints.append((base_path, row, col, lint, descr, linter))
+
+    lints_per_linter = dict()
+    for base_path, row, col, lint, descr, linter in lints:
+        if linter not in lints_per_linter:
+            lints_per_linter[linter] = []
+        lints_per_linter[linter].append((base_path, row, col, lint, descr))
+
+    ## Uncomment to get the number of lints per linter
+    # for linter, lints in lints_per_linter.items():
+    #     print(f"{linter}: {len(lints)}")
+
+    return lints_per_linter
+
+
+def filter_lints(lints_per_linter, filter_team: str=None, filter_linters: str=None):
+    """
+    Keeps only the lints from a specific team and specific linters.
+
+        Parameters:
+            filter_team (str): Keep only the lints from the files owned by the filter_team. None will keep lints from all teams.
+            filter_linters (str): Comma-separated linters to keep. None will keep lints from all linters.
+    """
+    list_filter_linters = filter_linters.split(',') if filter_linters else []
+    filtered_lints = dict()
+    codeowners = read_owners(CODEOWNERS_FILE_PATH)
+    for linter in lints_per_linter:
+
+        # If either we didn't set a filter or the linter is in the filter list
+        if not filter_linters or linter in list_filter_linters:
+            if linter not in filtered_lints:
+                filtered_lints[linter] = []
+
+            if not filter_team:
+                filtered_lints[linter] = lints_per_linter[linter]
+            else:
+                # Filter only the lints owned by the filter_team
+                for lint in lints_per_linter[linter]:
+                    if codeowners.of(lint[0]) == []:
+                        continue
+                    team_owning_the_lint = codeowners.of(lint[0])[0][1]
+                    if team_owning_the_lint == filter_team:
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I think something like that should work:
```suggestion
                    if ('TEAM', filter_team) in codeowners.of(lint[0]):
```

Reply:
Right ! Fixed in [8a4ad8d](https://github.com/DataDog/datadog-agent/pull/19015/commits/8a4ad8df6bc15855388760fd383eb8ca25948ab0).